state.go

```diff
@@ -835,6 +834,18 @@ func (cw *CheckWaitState) Handle(ctx *StateContext, c Controller) (State, bool)
 		return cw.Wait(next.state, cw, wait)
 	}
 
+	// (MEN-2195): Set the last udpate/inventory check time to now, as an error in an enter script will
+	// hinder these states from ever running, and thus causing an infinite loop if the script
+	// keeps returning the same error.
+	switch (next.state).(type) {
+	case *InventoryUpdateState:
+		log.Info("Inventory update state next!")
```

Comment:
Moved the update of the `lastInventoryUpdate` and `lastUpdateCheck' to here, and renamed them, as they are now more semantically like attemps (although they functioned like this previously as well, only the time was set upon reaching the state. Still it was oblivious to whether or not the action was successful or not).